boards: snps: nsim: align ICCM/DCCM configuration

tagunil opened the pull request:
Some of the boards are completely simulated and do not represent any real hardware configurations. Others are diverged from the original intentions. Treat them on a case-by-case basis, separately for EM, SEM and HS.

Original description:
> As these boards are completely simulated and do not represent any real hardware configurations, provide all of them with 1M of both ICCM and DCCM to avoid out-of-memory test failures and fix all discovered inconsistencies in their ICCM/DCCM configuration.

ruuddw commented:
Wasn't the idea that these nSIM boards matched some of our hardware development boards/emulation environments, so a single binary could be used in both simulation and hardware environment?

tagunil commented:
> Wasn't the idea that these nSIM boards matched some of our hardware development boards/emulation environments, so a single binary could be used in both simulation and hardware environment?

If so, I'm afraid this correspondence was already lost before my times, except maybe for some boards like HS5/HS6 which I haven't touched in this PR. Still, I think I could match all the others back again if I had some singular source of truth about the hardware configurations we want to simulate. Otherwise I believe it's better to have internally consistent configurations, even if unrealistic.

tagunil commented:
Hi @ruuddw and @evgeniy-paltsev, let's decide how to proceed. I've just reverted VPX5 back to 256K as requested. Do you have any other configurations you have hard data about, or we could have this quick fix and analyse the configurations later on a case-by-case basis?

tagunil committed:
boards: snps: nsim: align SEM platforms

Fix ICCM/DCCM properties in DTS to match the intended configuration,
along with some other minor inconsistencies.

Signed-off-by: Ilya Tagunov <Ilya.Tagunov@synopsys.com>

tagunil committed:
boards: snps: nsim: align HS platforms

As these boards are completely simulated and do not represent any real
hardware configurations anymore, provide them with 1M of both ICCM and
DCCM to avoid any out-of-memory test failures.

Signed-off-by: Ilya Tagunov <Ilya.Tagunov@synopsys.com>

tagunil committed:
boards: snps: nsim: align EM7D_v22 platform

Fix ICCM/DCCM properties both in DTS and in MDB settings to match the
original hardware configuration.

Signed-off-by: Ilya Tagunov <Ilya.Tagunov@synopsys.com>

ruuddw reviewed (approved):
Thanks, this aligns with other boards where needed and fixes the inconsistencies.
